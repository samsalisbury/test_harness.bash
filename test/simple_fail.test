#!/usr/bin/env bash

set -euo pipefail

# shellcheck disable=SC1090
source "${TESTING_BASH:-../testing.bash}"

fatalhelper() {
  helper
  fatal "This line number is where fatalhelper was called."
}

errorhelper() {
  helper
  error "This line number is where errorhelper was called."
}

TestFail_false() {
  log "You will see this."
  false
  log "You won't see this."
}

TestFail_mustrun_false() {
  log "You will see this."
  mustrun false
  log "You won't see this."
}

TestFail_error() {
  log "You will see this."
  error "some error"
  log "You will see this."
}

TestFail_fatal() {
  log "You will see this."
  fatal "some fatal error"
  log "You won't see this."
}

TestFail_fatalhelper() {
  log "You will see this."
  fatalhelper
  log "You won't see this."
}

TestFail_errorhelper() {
  log "You will see this."
  errorhelper
  log "You will see this."
}
