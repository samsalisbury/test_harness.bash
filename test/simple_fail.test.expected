=== RUN   simple_fail/TestFail_false
--- FAIL: simple_fail/TestFail_false
    simple_fail.test:19: You will see this.
    simple_fail.test:20: Command failed with exit code 1: false
=== RUN   simple_fail/TestFail_mustrun_false
--- FAIL: simple_fail/TestFail_mustrun_false
    simple_fail.test:25: You will see this.
    $ false
    simple_fail.test:26: Command failed with exit code 1
=== RUN   simple_fail/TestFail_error
--- FAIL: simple_fail/TestFail_error
    simple_fail.test:31: You will see this.
    simple_fail.test:32: some error
    simple_fail.test:33: You will see this.
=== RUN   simple_fail/TestFail_fatal
--- FAIL: simple_fail/TestFail_fatal
    simple_fail.test:37: You will see this.
    simple_fail.test:38: some fatal error
=== RUN   simple_fail/TestFail_fatalhelper
--- FAIL: simple_fail/TestFail_fatalhelper
    simple_fail.test:43: You will see this.
    simple_fail.test:44: This line number is where fatalhelper was called.
=== RUN   simple_fail/TestFail_errorhelper
--- FAIL: simple_fail/TestFail_errorhelper
    simple_fail.test:49: You will see this.
    simple_fail.test:50: This line number is where errorhelper was called.
    simple_fail.test:51: You will see this.
FAIL
fail      simple_fail
